### PR TITLE
JP-3223: Extract_2d 'slit_name' parameter fails for NRS_MSASPEC slits

### PR DESCRIPTION
**Pull request by penaguerrero**
Resolves [JP-3223](https://jira.stsci.edu/browse/JP-3223)  (again)

Closes #

This PR addresses a crash of extract_2d when user provides slit_name, and value is a string. This change had been done and merged in #8108 , but it accidentally got undone by a later PR. So note that the original fix was NOT included in DMS B10.1 as we'd assumed. It will now be included in B10.2.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -42,7 +42,13 @@ extract_1d
 
 - Added a hook to bypass the ``extract_1d`` step for NIRISS SOSS data in 
   the FULL subarray with warning. [#8225]
-  
+
+extract_2d
+----------
+
+- Fixed crash when user provides slit_name, and value is a string. This
+  change had been done in PR8108 but it got undone by another PR. [#8272]
```
Comment (hbushouse) [inline CHANGES.rst:50]:
```suggestion
- Fixed crash when user provides an integer value for the `slit_name` argument,
  by converting to a string. This change had been done in #8108, but it got undone
  by another PR. [#8272]
```

**Review by hbushouse — COMMENTED** [review]
Needs a regression test run.

**Comment by penaguerrero** [comment]
reg tests running at: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1201/

**Comment by penaguerrero** [comment]
Reg tests succeeded! Only failures are due to unable to retrieve file from Artifactory.

**Review by hbushouse — APPROVED** [review]

**Pull request merged**